packages/protocol/contracts/L1/libs/LibTokenomics.sol

```diff
@@ -180,26 +180,23 @@ library LibTokenomics {
         TaikoData.FeeConfig memory feeConfig,
         uint256 feeBase,
         bool isProposal,
-        uint256 tNow, // seconds
-        uint256 tLast, // seconds
-        uint256 tAvg // milliseconds
-    ) internal pure returns (uint256 newFeeBase, uint256 tRelBp) {
-        if (tAvg == 0 || tNow == tLast) {
+        uint256 timeUsed, // seconds
+        uint256 timeAverage // milliseconds
+    ) internal pure returns (uint256 newFeeBase, uint256 premiumRate) {
+        if (timeAverage == 0) {
             return (feeBase, 0);
         }
-
         unchecked {
-            tAvg = tAvg.min(feeConfig.avgTimeCap);
-            uint256 max = (feeConfig.maxPeriodPctg * tAvg) / 100;
-            uint256 grace = (feeConfig.gracePeriodPctg * tAvg) / 100;
-            uint256 t = ((tNow - tLast) * 1000).max(grace).min(max);
-            tRelBp = (10000 * (t - grace)) / (max - grace); // [0-10000]
-            uint256 alpha = 10000 + (tRelBp * feeConfig.multiplerPctg) / 100;
+            uint p = feeConfig.startBips; // [0-10000]
+            uint a = timeAverage;
+            uint t = (timeUsed * 1000).min(a * 2); // millisconds
+
+            newFeeBase = (feeBase * (10000 + (t * p) / a - p)) / 10000;
```

Comment:
I still find this formula very tricky to really understand. If not using one of the other common formula's to calculate something like, why can't this be as simple as something like this (disregarding rounding problems, so needs some additional factors):
```
newFeeBase = feeBase * ((timeUsed / timeAverage) / SMOOTHING_FACTOR);
```

Reply:
Brecht, I think the implementation is exactly the same as the one you suggested.
`(t * p) / a - p)` = `p * (t/a - 1)`

`(10000 + (t * p) / a - p)) / 10000` = `(1 + p * (t/a - 1))`

p is the DAMPING_FACTOR, t is timUsed, a is timeAvareege.

Reply:
Ah haha didn't realize they were that close. Why is it the damping factor called `startBips` (I though it was also used for something else, didn't know why it was also used here)?

Reply:
startBips is not a good name, `dampingFactor` instead? It is only used here in the time-based fee calculation.